storage/storage.go

```diff
@@ -23,35 +26,34 @@ type ObjectInfo struct {
 type Storage interface {
 	// Stat can be used to find information about the object corresponding to the given element.
 	// If the object does not exist, then Stat returns an error satisfied by os.IsNotExist.
-	Stat(ctx context.Context, element ingest.Identifiable) (*ObjectInfo, error)
-	Store(ctx context.Context, element ingest.Identifiable, download func(context.Context, ingest.Identifiable) (ingest.Object, error)) (*url.URL, error)
+	Stat(ctx context.Context, element ingest.SimpleCodec) (*ObjectInfo, error)
+	Store(ctx context.Context, element ingest.SimpleCodec, obj ingest.Object) (*url.URL, error)
+	Configure(map[string]any) error
```

Comment:
This isn't actually relevant to this interface, is it? It's just a sneaky way to get the plugin interface implemented?

Reply:
Yes and I think we should get rid of the Storage interface and only implement the Destination interface. Otherwise the package that will implement the Configure interface needs access to the Destination implementation's fields in some way.

Of course we could do something like 
```go
type destinationImp struct {
 d Destination
}

func (d * destinationImp) Configure()error {
  ... 
  d.d = drive.New(...)
}

func (d *destinationImp) Stat(...) ...{
  return d.d.Stat(...
  
```

I am not sure if this is a useful abstraction

Reply:
If we do want to get rid of our `Storage` interface then we are essentially getting rid of ingest as a library and saying that it should only be used in plugin-fashion, right? That's not bad, I just want to be clear about it. It means that we can get rid of lots of abstractions and lots of of documentation for how to use the project I think

Reply:
I see when we use ingest as library we would like the storage interface, mainly the S3 implementation